modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -274,6 +277,17 @@ public ObjectField addSystemObjectField(
 			dbColumnName = name;
 		}
 
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
+
+		if (objectDefinition.isModifiable() &&
+			!ObjectFieldUtil.isMetadata(name) &&
+			!SystemObjectDefinitionManagementChecker.isInvokerBundleAllowed()) {
+
+			throw new ObjectFieldSystemException(
+				"Only allowed bundles can add system object fields");
+		}
```

Comment:
I think we can remove this condition ` !ObjectFieldUtil.isMetadata(name)` because this is already checked in _addObjectField().
Then, if this happens, the exception will be more specific to the user :+1:

Reply:
We talked and decided to add the `objectDefinition.isSystem()` instead the `!ObjectFieldUtil.isMetadata(name)`